File: split/openorclosein.pas
procedure openorclosein;
var
  c: 0..1;
  n: 0..15;
begin
  c := curchr;
  scanfourbitint;
  n := curval;
  if readopen[n] <> 2 then
  begin
    aclose(readfile[n]);
    readopen[n] := 2;
  end;
  if c <> 0 then
  begin
    scanoptionalequals;
    scanfilename;
    if curext = 338 then
      curext := 790;
    packfilename(curname, curarea, curext);
    if aopenin(readfile[n]) then
      readopen[n] := 1;
  end;
end;

File: cpp/openorclosein.cpp
#include "openorclosein.h"
#include "scanfourbitint.h"
#include "aclose.h"
#include "scanoptionalequals.h"
#include "scanfilename.h"
#include "packfilename.h"
#include "aopenin.h"
#include "texte.h"

void openorclosein(void)
{
	auto c = curchr;
	scanfourbitint();
	auto n = curval;
	if (readopen[n] != closed)
	{
		aclose(readfile[n]);
		readopen[n] = closed;
	}
	if (c)
	{
		scanoptionalequals();
		scanfilename();
		if (curext == "") 
			curext = ".tex";
		pack_cur_name();
		if (aopenin(readfile[n]))
			readopen[n] = just_open;
	}
}
